Tests/UnitTests/Purchasing/Purchases/PurchasesLogInTests.swift

```diff
@@ -113,6 +113,30 @@ class PurchasesLogInTests: BasePurchasesTests {
         expect(self.identityManager.invokedSwitchUserParametersList) == ["test-user-id"]
     }
 
+    func testRefreshesOfferingsCache() {
+        expect(self.mockOfferingsManager.invokedUpdateOfferingsCacheCount) == 0
+
+        self.systemInfo = MockSystemInfo(finishTransactions: true, customEntitlementsComputation: true)
+        Purchases.clearSingleton()
+        self.initializePurchasesInstance(appUserId: "old-test-user-id")
+
+        self.purchases.switchUser(to: "test-user-id")
+
+        expect(self.mockOfferingsManager.invokedUpdateOfferingsCacheCount).toEventually(equal(1))
```

Comment:
eventually because we need to check if the app is running in the background and that ironically requires a thread change IIRC

Reply:
Not with the mock though? Not a big deal anyway.

Reply:
ohh right, maybe? I didn't check because I couldn't run this 😅